### DISCUSSION TITLE
Differentiate action on selected text based on the window name / process owning window where the text is selected.

### DISCUSSION THREAD

**Discussion by GOPiGal** [hluk/CopyQ#2899] [Ideas]
Hello,

May I propose a new feature where it is possible to customize action on selected text depends on where the selection happens?
My use case is simple and can be use as an example:
1.  if I select text outside of terminal nothing happens
2. if I select text inside the terminal it is automatically copied (stored in CopyQ Tab)

Other use cases could be applying some cleaning, reformation, translation, etc based on the  fact where the text was selected (e.g. copied text might need different action depends if it comes from Code Visual might be different from InteliJ or Firefox)

Also it does not not necessarily need to be a text, could be an image or any other supported object, though I supose text is the most important...

Cheers,
Piotr

**Comment by hluk** [comment]
This is already supported via custom commands. For example, see [Copy Clipboard to Window Tabs](https://github.com/hluk/copyq-commands/tree/master/Automatic#copy-clipboard-to-window-tabs) (see [README.md](https://github.com/hluk/copyq-commands/blob/master/README.md) for details on how to add the command).

**Reply by GOPiGal** [reply]
Either I do not how to do it or you have misunderstood me. How to check when is any type terminal or any other window. Also the idea is that if it is outside of the terminal nothing (like nothing, nothing) happens and previous content of the clipboard remains untouched...

**Reply by hluk** [reply]
CopyQ can only distinguish windows by their window title.

For example, here is a command to abort processing a copied clipboard content unless the window title matches a regular expression (`Terminal`):

```ini
[Command]
Automatic=true
Command="
    copyq:
    const regexToMatchWindowTitle = /^Terminal$/;
    const windowTitle = str(data(mimeWindowTitle));
    const windowMatches = regexToMatchWindowTitle.test(windowTitle);
    if (!windowMatches) {
        ignore();
    }"
Name=Store Only Terminal
```

**Reply by GOPiGal** [reply]
Hello,
First - thank you for replies! Greatly appreciated!!!

That is not entirely what  I am looking for. The reason is that if I press [Ctrl]+[C] in any other window than the Terminal's one I want CQ to pick it up and, if I understand it correctly, will not happen here. My idea is only about the copying that happen when you select text. In short it is to replicate the behavior from Windows. 

So in other words:

Terminal window:
1. Text Selected with mouse is automatically copied and synchronized in both clipboards
2. Text Selected and [Ctr]+[Shift]+[C] is copied and synchronized in both clipboards

Any other Window:
1. Text Selected with mouse is **NOT** automatically copied and synchronized in both clipboards (no action taken)
2. Text Selected and [Ctr]+[C] is copied and synchronized in both clipboards

Cheers,
Piotr
